src/main/java/org/pentaho/di/trans/dataservice/optimization/OptimizationImpactInfo.java

```diff
@@ -76,7 +78,7 @@ public String getDescription() {
       .append( "Step:  " )
       .append( getStepName() )
       .append( "\n" );
-    if ( getErrorMsg().length() > 0 ) {
+    if ( StringUtil.isEmpty( getErrorMsg() ) ) {
       builder
```

Comment:
Hey @denis8703 -  my last comment was: 
    assume that an empty or null error message means "no error".

Your change does the opposite-- an empty or null error message will result in an error message being appended to the output, and will skip adding the error message if present.